### PR TITLE
Move generated app to use Jest as test solution

### LINKED ISSUE #447: Move generated app to use Jest as test solution
Following #406. We can remove all other dependencies and use also Jest in generated app (recommended by community and used it in React and Redux projecst). User can also choose to use Snapshots if they want too (maybe we could pass a flag to generate snapshot test if they want to).

This will also fix #333 since we won't need istanbul anymore.

This depends on following PR: #437 & #445

### PR DESCRIPTION
**Pull request by ferrannp**
Fixes #447, #333, #452.

- [x] Modify `test.js` to use Jest
- [x] Configure Jest to detect webpack aliases in tests
- [x] Handle arguments (`--coverage`, `--watch`,`--pattern`)
- [x] Handle `--debug-test` option
- [x] Delete all related to `karma`, `istanbul`, `mocha`, etc

### CODE REVIEW

File: src/commands/test.js [hunk 1]
```diff
@@ -1,171 +1,53 @@
...
-    args.push("--debug-brk");
-  }
-  if (!options.single && !options.debugTest) {
-    args.push("--watch");
-  }
-  return args;
-}
+  // We map webpack aliases from webpack-isomorphic-tools-config file so Jest can detect them in tests too
+  Object.keys(alias).forEach((key) => {
+    moduleNameMapper[`^${key}(.*)$`] = `${alias[key]}$1`;
```
Comment (toddw) [inline src/commands/test.js:15]: :+1:

**Comment by toddw** [comment]
This is looking great so far

**Comment by toddw** [comment]
We should merge in the changes from https://github.com/TrueCar/gluestick/pull/458 and update the predefined generators but I don't think that should be too much work

**Comment by toddw** [comment]
I just tested out `--coverage`, `--watch` and `--pattern` and they worked great.

Also tested out `--debug-test` and it worked great too! Very glad we don't even have to pass a filename to it! 👍

**Comment by ferrannp** [comment]
This is ready to be merged. @toddw I tried to make sure to remove all old unused testing code but reviewer might want to double check :)

**Comment by toddw** [comment]
Looks good to me. Please resolve merge conflict and fee free to merge. Thanks!

**Pull request merged**